### PR TITLE
[ISSUE#5695] Add a plugin SPI for configuration encryption and decryption for Nacos 2.0

### PR DESCRIPTION
**Pull request by li-xiao-shuang**
#5693 

Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: config/src/main/resources/META-INF/nacos-db.sql [hunk 1]
```diff
@@ -153,6 +154,7 @@ CREATE TABLE `his_config_info` (
   `src_ip` varchar(20) DEFAULT NULL,
   `op_type` char(10) DEFAULT NULL,
   `tenant_id` varchar(128) DEFAULT '' COMMENT '租户字段',
+  `secret_key` text NOT NULL COMMENT '秘钥',
```
Comment (KomachiSion) [inline config/src/main/resources/META-INF/nacos-db.sql:157]: why not use name `encrypted_data_key`?

In your code, all relative named `encryptedDataKey` and `ENCRYPTED_DATA_KEY`, but in db is `secret_key`?

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Review by KomachiSion — CHANGES_REQUESTED** [review]
I suggest you divide pr into 2~3 parts
1. The basic content of the plug-in (such as the spi part)
2. The corresponding part of the server
3. The corresponding part of the client

File: config/src/main/resources/META-INF/nacos-db.sql (hunk 1 above)
Comment (li-xiao-shuang) [reply config/src/main/resources/META-INF/nacos-db.sql:157]: encryptedDataKey is the original，But I think secret_key is more accurate

Comment (li-xiao-shuang) [reply config/src/main/resources/META-INF/nacos-db.sql:157]: > why not use name `encrypted_data_key`?
> 
> In your code, all relative named `encryptedDataKey` and `ENCRYPTED_DATA_KEY`, but in db is `secret_key`?

encryptedDataKey is the original，But I think secret_key is more accurate

**Comment by li-xiao-shuang** [comment]
> I suggest you divide pr into 2~3 parts
> 
> 1. The basic content of the plug-in (such as the spi part)
> 2. The corresponding part of the server
> 3. The corresponding part of the client

Currently this PR can be combined，I'm updating on the secret key

File: api/src/main/java/com/alibaba/nacos/api/config/CryptoExecutor.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     * Execute decryption.
+     *
+     * @param dataId    dataId
+     * @param secretKey secretKey
+     * @param content   content
+     * @return decrypt value
+     */
+    public static String executeDecrypt(String dataId, String secretKey, String content) {
+        CryptoSpi cryptoSpi = cryptoInstance(dataId);
+        if (Objects.isNull(cryptoSpi)) {
```
Comment (pirateskipper) [inline api/src/main/java/com/alibaba/nacos/api/config/CryptoExecutor.java:57]: `return cryptoSpi == null ? content : cryptoSpi.decrypt(secretKey, content);`

File: api/src/main/java/com/alibaba/nacos/api/config/CryptoExecutor.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    
+    /**
+     * Parse prefix match encryption algorithm.
+     *
+     * @param dataId dataId
+     * @return Encryption algorithm instance
+     */
+    public static CryptoSpi cryptoInstance(String dataId) {
+        boolean result = checkCipher(dataId);
+        if (result) {
```
Comment (pirateskipper) [inline api/src/main/java/com/alibaba/nacos/api/config/CryptoExecutor.java:71]: `if (checkCipher(dataId)) {`

Comment (li-xiao-shuang) [reply api/src/main/java/com/alibaba/nacos/api/config/CryptoExecutor.java:71]: > `if (checkCipher(dataId)) {`
👌🏻

File: api/src/main/java/com/alibaba/nacos/api/config/CryptoExecutor.java (hunk 2 above)
Comment (li-xiao-shuang) [reply api/src/main/java/com/alibaba/nacos/api/config/CryptoExecutor.java:57]: > `return cryptoSpi == null ? content : cryptoSpi.decrypt(secretKey, content);`

👌🏻

**Review by KomachiSion — APPROVED** [review]

File: config/src/main/resources/META-INF/nacos-db.sql (hunk 1 above)
Comment (Futub67) [reply config/src/main/resources/META-INF/nacos-db.sql]: Let`s chat us  tomorrow

**Pull request merged**